geopandas/geodataframe.py

```diff
@@ -1456,29 +1483,45 @@ def explode(self, column=None, **kwargs):
             column = self.geometry.name
         # If the specified column is not a geometry dtype use pandas explode
         if not isinstance(self[column].dtype, GeometryDtype):
-            return super(GeoDataFrame, self).explode(column, **kwargs)
-            # TODO: make sure index behaviour is consistent
+            if compat.PANDAS_GE_11:
+                return super(GeoDataFrame, self).explode(column, ignore_index)
+            else:
+                return super(GeoDataFrame, self).explode(column)
 
         df_copy = self.copy()
 
         if "level_1" in df_copy.columns:  # GH1393
             df_copy = df_copy.rename(columns={"level_1": "__level_1"})
 
-        exploded_geom = df_copy.geometry.explode().reset_index(level=-1)
-        exploded_index = exploded_geom.columns[0]
+        if add_multiindex:
+            exploded_geom = df_copy.geometry.explode()
+            exploded_index = exploded_geom.index
+            exploded_geom = exploded_geom.reset_index(level=-1).drop("level_1", axis=1)
+        else:
+            exploded_geom = (
+                df_copy.geometry.explode().reset_index(level=-1).drop("level_1", axis=1)
+            )
+            exploded_index = exploded_geom.index
 
         df = pd.concat(
             [df_copy.drop(df_copy._geometry_column_name, axis=1), exploded_geom], axis=1
         )
         # reset to MultiIndex, otherwise df index is only first level of
         # exploded GeoSeries index.
-        df.set_index(exploded_index, append=True, inplace=True)
-        df.index.names = list(self.index.names) + [None]
-
+        # df.set_index(exploded_index, append=True, inplace=True)
+        df.index = exploded_index
```

Comment:
I would prefer to keep the `set_index` method rather than directly assign it. Is there a reason why you commented that option out? (the comment should be cleaned anyway)

Reply:
I think it was because i did not manage to get the same result with `set_index`. Now I have tried a bit more, and the problem was `append=True`. Since in my case now I'm computing the entire index again, I don't need to append it.

Out of curiosity, what makes you prefer `set_index` ?